### PR TITLE
router module update

### PR DESCRIPTION
**Pull request by AnittaPattu**

### CODE REVIEW

File: src/components/Authorization/components/Login/login.jsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    setemail(data);
+  };
+  const passwordData = (data) => {
+    setPassword(data);
+  };
+
+  function login() {}
+
+  async function formData() {
+    var data = {
```
Comment (SomeshThakur) [inline src/components/Authorization/components/Login/login.jsx]: Don't use var

File: src/components/Authorization/components/Login/login.jsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    };
+    const requestOptions = {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(data),
+    };
+    const response = await fetch(loginUrl, requestOptions);
+    fetch(loginUrl, requestOptions)
+      .then((response) => {
+        console.log(response);
```
Comment (SomeshThakur) [inline src/components/Authorization/components/Login/login.jsx]: You can remove console.log

Comment (SomeshThakur) [reply src/components/Authorization/components/Login/login.jsx]: Or remove this fetch, as its not doing anything

File: src/components/CreateCourse/createCourse.jsx [hunk 3]
```diff
@@ -52,16 +64,20 @@ function CreateCourse(props) {
       (authData) => authData.id !== authorDat.id
     );
     setCourseAuthor(courseAuth);
-    updateAuthorArray(courseAuth, "delete");
+    // updateAuthorArray(courseAuth, "delete");
```
Comment (SomeshThakur) [inline src/components/CreateCourse/createCourse.jsx]: remove committed code

File: .husky/pre-commit [hunk 4]
```diff
@@ -1,4 +0,0 @@
-#!/usr/bin/env sh
```
Comment (SomeshThakur) [inline .husky/pre-commit:1]: You should add pre-commit hook as per module, you can add linting & formatting may be

File: src/components/Authorization/components/Login/login.jsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
+import { useState } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import Button from "../../../../common/Button/button";
+import Input from "../../../../common/Input/input";
+
+export const Login = (props) => {
```
Comment (SomeshThakur) [inline src/components/Authorization/components/Login/login.jsx]: As you are not using TS, you should use PropTypes to validate props

File: src/components/Courses/components/course-info/course-info.jsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    if (
+      localStorage.getItem("token") === undefined ||
+      localStorage.getItem("token") === null
+    ) {
+      navigate("/login", { replace: true });
+    }
+  }, []);
+
+  return (
+    <>
```
Comment (SomeshThakur) [inline src/components/Courses/components/course-info/course-info.jsx]: You can remove this fragment

File: src/components/Courses/components/course-card/course-content.jsx [hunk 7]
```diff
@@ -1,15 +1,26 @@
...
 import { getCourseDuration } from "../../../../helper/getCourseDuration";
 import Button from "../../../../common/Button/button";
 import { getFormattedDate } from "../../../../helper/formatCreationDate";
+import { useNavigate } from "react-router-dom";
 
 const CourseCard = (props) => {
   const AUTHOR = "Author: ";
   const DURATION = "Duration: ";
   const CREATED = "Created: ";
   const BUTTON_TEXT = "Show Courses";
```
Comment (SomeshThakur) [inline src/components/Courses/components/course-card/course-content.jsx:12]: `Show course`

File: src/components/Authorization/components/Login/login.jsx [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    const response = await fetch(loginUrl, requestOptions);
+    fetch(loginUrl, requestOptions)
+      .then((response) => {
+        console.log(response);
+        return response.json();
+      })
+      .then((data) => console.log(data));
+    const loginData = await response.json();
+    if (loginData.successful) {
+      navigate("/login", { replace: true });
```
Comment (SomeshThakur) [inline src/components/Authorization/components/Login/login.jsx]: You are not saving token local Storage

File: src/components/Header/header.jsx [hunk 9]
```diff
@@ -1,17 +1,43 @@
...
+  const name = "";
+
+  useEffect(() => {
+    if (location.pathname.includes("course")) {
+      hideName = false;
+    }
+    if (localStorage.getItem("token")) {
+      name = localStorage.getItem("token").user.name;
+    }
+  }, []);
```
Comment (SomeshThakur) [inline src/components/Header/header.jsx:21]: I think this should have dependency of location

File: src/components/Header/header.jsx [hunk 10]
```diff
@@ -1,17 +1,43 @@
...
+  const navigate = useNavigate();
+  let hideName = true;
+  const name = "";
+
+  useEffect(() => {
+    if (location.pathname.includes("course")) {
+      hideName = false;
+    }
+    if (localStorage.getItem("token")) {
+      name = localStorage.getItem("token").user.name;
```
Comment (SomeshThakur) [inline src/components/Header/header.jsx]: You will get string from localStorage right ? How is it handling user.name?

File: src/components/Header/header.jsx [hunk 11]
```diff
@@ -1,17 +1,43 @@
...
-function Header(props) {
+function Header() {
   const HEADER_BUTTON_TEXT = "Logout";
+  const location = useLocation();
+  const navigate = useNavigate();
+  let hideName = true;
+  const name = "";
+
+  useEffect(() => {
+    if (location.pathname.includes("course")) {
```
Comment (SomeshThakur) [inline src/components/Header/header.jsx]: I think you should check if page is login/registration.

**Review by SomeshThakur — CHANGES_REQUESTED** [review]
Let me know if you need help in addressing any of the comments.

File: src/components/Authorization/components/Login/login.jsx (hunk 2 above)
Comment (AnittaPattu) [reply src/components/Authorization/components/Login/login.jsx]: Done

File: src/components/CreateCourse/createCourse.jsx (hunk 3 above)
Comment (AnittaPattu) [reply src/components/CreateCourse/createCourse.jsx]: Removed the commented code

File: src/components/Authorization/components/Login/login.jsx (hunk 5 above)
Comment (AnittaPattu) [reply src/components/Authorization/components/Login/login.jsx]: updated proptypes

File: .husky/pre-commit (hunk 4 above)
Comment (SomeshThakur) [reply .husky/pre-commit:1]: To be addressed in next MR

**Pull request merged**